packages/apps/api/src/app.ts

```diff
@@ -1,69 +1,119 @@
-import express, {Application, Request, Response} from 'express';
+import Auth0Client from '@weco/auth0-client';
+import SierraClient from '@weco/sierra-client';
 import bodyParser from 'body-parser';
-import SierraClient from "@weco/sierra-client";
-import {validateCredentials} from "./handlers/auth";
-import {DummyUserOne, DummyUserTwo} from "./models/user";
-import {getUser} from "./handlers/user";
-import Auth0Client from "@weco/auth0-client";
+import cors from 'cors';
+import express, { Application, Request, Response } from 'express';
+import { validateCredentials } from './handlers/auth';
+import { getUser } from './handlers/user';
+import { DummyUserOne, DummyUserTwo } from './models/user';
+
+export default createApplication();
 
-const app: Application = express();
 const sierraClient: SierraClient = new SierraClient(
   process.env.SIERRA_API_ROOT!, process.env.SIERRA_CLIENT_KEY!, process.env.SIERRA_CLIENT_SECRET!
 );
+
 const auth0Client: Auth0Client = new Auth0Client(
   process.env.AUTH0_API_ROOT!, process.env.AUTH0_API_AUDIENCE!, process.env.AUTH0_CLIENT_ID!, process.env.AUTH0_CLIENT_SECRET!
 );
 
-app.use(bodyParser.json());
-app.use(bodyParser.urlencoded({extended: true}));
-
-app.post('/auth', (req, res) => validateCredentials(sierraClient, req, res));
+function createApplication(): Application {
+  const app: Application = express();
 
-app.get('/users', function (req: Request, res: Response) {
-  console.log('Processing [GET /users]...');
-  res.status(200).json([DummyUserOne, DummyUserTwo]);
-});
+  app.use(bodyParser.json());
+  app.use(bodyParser.urlencoded({ extended: true }));
```

Comment:
You can use `app.use(...)` on your CORS calls too so you can centralize them here:

```js
const corsOptions: cors.CorsOptions = {
   allowedHeaders: ['Content-Type', 'X-API-Key'],
   methods: 'OPTIONS,POST',
   origin: process.env.API_ALLOWED_ORIGINS
};

app.use(cors(corsOptions));
```

Reply:
Yes indeed and that was how it was during the initial testing, but some of the resources accept different headers and different methods. If we used the catch-all `app.use(...)` approach, then e.g.

```
[OPTIONS] /auth
```

would return `GET, PUT, DELETE, POST` for the available methods and `X-API-Key` and `Authorization` for the acceptable methods, whereas this endpoint actually only supports `POST` and requires the `X-API-Key`. By breaking out each resource into its own CORS configuration, this allows us to specific about what methods and heads that exact resource path supports.

```
app.options('*', cors()) // include before other routes
```

This would never actually be called because the API Gateway that sits above the API Lambda doesn't have the concept of a wildcard path, so any query to any resource path that isn't explicitly created in API Gateway and subsequently reflected in this Lambda's mapping would just return a 404 by the API Gateway and wouldn't make it this far.